### PR TITLE
Returns the cached version only when network request fails

### PR DESCRIPTION
**Pull request by GreatWizard**
## Changes proposed in this pull request

I add a new option `fallback` to activate this feature.
It returns cached version only when network fails else it fetch the index.

### CODE REVIEW

**Comment by eshtadc** [comment]
Does this duplicate what is available in https://github.com/DockYard/ember-service-worker-cache-fallback?

**Comment by olegstepura** [comment]
> Does this duplicate what is available in https://github.com/DockYard/ember-service-worker-cache-fallback?

No, this PR tries to achieve another caching strategy for `index.html`:

## currently:

- index plugin handles `INDEX_HTML_URL`
- if that url is in the cache - it returns it from cache
- then it loads new version and fills the cache.

This will not work for example with [Play framework flash messaging](https://www.playframework.com/documentation/2.7.x/ScalaSessionFlash) in a following scenario: 
when a message is to be flashed with a POST request - it will add cookie to be consumed by the following GET request (if POST does a redirect to GET). But here comes a surprise: sw returns cached version, not from server which does not use cookie value.

## this PR
- adds another strategy - load from server first, then fallback to cache if there was problem loading index.

it can solve the issue described above.

I would vote for merging this.

File: service-worker/index.js [hunk 1]
```diff
@@ -36,8 +45,15 @@ self.addEventListener('fetch', (event) => {
   let scopeIncluded = !INDEX_INCLUDE_SCOPE.length || urlMatchesAnyPattern(request.url, INDEX_INCLUDE_SCOPE);
 
   if (isGETRequest && isHTMLRequest && isLocal && scopeIncluded && !scopeExcluded) {
-    event.respondWith(
-      caches.match(INDEX_HTML_URL, { cacheName: CACHE_NAME })
-    );
+    if (INDEX_FALLBACK) {
+      event.respondWimth(
```
Comment (olegstepura) [inline service-worker/index.js]:
```suggestion
      event.respondWith(
```

**Comment by eshtadc** [comment]
This can now be handled by using the `strategy` option of `fallback` per https://github.com/DockYard/ember-service-worker-index/pull/17
It will fall back to cache either in case of timeout or fetch error.

**Pull request closed without merging**